src/cpp/r/R/Tools.R

```diff
@@ -1290,6 +1290,113 @@ environment(.rs.Env[[".rs.addFunction"]]) <- .rs.Env
    envirs
 })
 
+.rs.addFunction("heredoc", function(text)
+{
+   # remove leading, trailing whitespace
+   trimmed <- gsub("^\\s*\\n|\\n\\s*$", "", text)
+   
+   # split into lines
+   lines <- strsplit(trimmed, "\n", fixed = TRUE)[[1L]]
+   
+   # compute common indent
+   indent <- regexpr("[^[:space:]]", lines)
+   common <- min(setdiff(indent, -1L))
+   paste(substring(lines, common), collapse = "\n")
+})
+
+.rs.addFunction("bugReport", function(pro = NULL)
+{
+   # collect information about the running version of R / RStudio
+   rstudioInfo <- .rs.api.versionInfo()
+   rstudioVersion <- format(rstudioInfo$version)
+   
+   rstudioEdition <- sprintf(
+      "%s [%s]",
+      if (rstudioInfo$mode == "desktop") "Desktop" else "Server",
+      if (is.null(rstudioInfo$edition)) "Open Source" else toupper(rstudioInfo$edition)
+   )
+   
+   rInfo <- utils::sessionInfo()
+   rVersion <- rInfo$R.version$version.string
+   rVersion <- sub("^R version", "", rVersion, fixed = TRUE)
+   osVersion <- rInfo$running
+   
+   rInfoText <- local({
+      op <- options(width = 78)
+      on.exit(options(op), add = TRUE)
+      paste(capture.output(print(rInfo)), collapse = "\n")
+   })
+   
+   # create issue template and fill in the pieces
+   template <- .rs.heredoc("
+      ### System details
+
+      RStudio Edition : %s
```

Comment:
Nit - doesn't look like these lines are going to get the monospace treatment so the `:` will get staggered. Maybe change the formatting so they get monospaced like our real bug template, or just remove all the spacing so it doesn't look like they're aligned?

Reply:
Good point; I'll make that change. Thanks!